File: 백준/Silver/10845. 큐/큐.cc
#include<iostream>
#include<vector>

using namespace std;

int main() {

	int N;
	vector<int> que;
	vector<int> sol;
	string str;
	int num;

	cin >> N;

	for (int i = 0; i < N; i++) {
		cin >> str;
		if (str == "push") {
			cin >> num;
			que.push_back(num);
		}
		else if (str == "pop") {
			if (que.size() == 0) {
				sol.push_back(-1);
			}
			else {
				sol.push_back(que[0]);
				que.erase(que.begin());
			}
		}
		else if (str == "size") {
			sol.push_back(que.size());
		}
		else if (str == "empty") {
			if (que.size() == 0) {
				sol.push_back(1);
			}
			else {
				sol.push_back(0);
			}
		}
		else if (str == "front") {
			if (que.size() == 0) {
				sol.push_back(-1);
			}
			else {
				sol.push_back(que[0]);
			}
		}
		else if (str == "back") {
			if (que.size() == 0) {
				sol.push_back(-1);
			}
			else {
				sol.push_back(que[que.size() - 1]);
			}
		}
	}

	for (int i = 0; i < sol.size(); i++) {
		cout << sol[i] << '\n';
	}

	return 0;
}

File: 백준/Silver/10845. 큐/큐.java
import java.io.*;
import java.util.ArrayDeque;
import java.util.Deque;


class Main{
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        Deque<Integer> deque = new ArrayDeque<>();
        int N = Integer.parseInt(br.readLine());

        for(int i=0; i<N; i++) {
            String[] input = br.readLine().split(" ");
            if (input[0].equals("push")) {
                deque.addLast(Integer.parseInt(input[1]));
            }
            else if (input[0].equals("pop")) {
                if(!deque.isEmpty()) {
                    bw.write( deque.removeFirst()+"\n");
                }
                else {
                    bw.write("-1\n");
                }
            }
            else if (input[0].equals("size")) {
                bw.write(deque.size()+"\n");
            }
            else if (input[0].equals("empty")) {
                if (deque.isEmpty()) {
                    bw.write(1+"\n");
                }
                else {
                    bw.write(0+"\n");
                }
            }
            else if (input[0].equals("front")) {
                if(!deque.isEmpty()) {
                    bw.write(deque.peekFirst()+"\n");
                }
                else {
                    bw.write("-1\n");
                }
            }
            else if (input[0].equals("back")) {
                if(!deque.isEmpty()) {
                    bw.write(deque.peekLast()+"\n");
                }
                else {
                    bw.write("-1\n");
                }
            }
        }

        bw.flush();
        bw.close();
    }
}